### ISSUE TITLE
Depending on alias is not transitive

### ISSUE THREAD

**Issue by ejgallego** [ocaml/dune#3415]
Hi folks,

I'm not sure if this intended but posting here for discussion. Assume a dune file:

```
(alias
 (name foo)
 (deps (source_tree dir)))

(rule
 (alias bar)
 (deps (alias foo))
 (action (with-stdout-to out (bash "cat dir/dep"))))
```
`dune build @bar` will build `out` correctly the first time; however it we update `dir/dep` `out` is not rebuilt.

Dune 2.5.1 (back to 2.1.1), OCaml 4.10, Linux

**Comment by jeremiedimino** [comment]
This looks like a bug indeed. Morally, depending on `(alias foo)` should depend on the expansion of `foo`, which should include `dir/dep` because of the `(deps (source_tree dir))`.

**Comment by aalekseyev** [comment]
This issue should be fixed by #4303.
